src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_huhuran.cpp

```diff
@@ -15,148 +15,144 @@
-    EMOTE_BERSERK               = 1,
-
-    SPELL_FRENZY                = 26051,
-    SPELL_BERSERK               = 26068,
-    SPELL_POISONBOLT            = 26052,
-    SPELL_NOXIOUSPOISON         = 26053,
-    SPELL_WYVERNSTING           = 26180,
-    SPELL_ACIDSPIT              = 26050,
-    SPELL_WYVERN_STING_DAMAGE   = 26233
+    EMOTE_BERSERK               = 1
 };
 
-struct boss_huhuran : public ScriptedAI
+enum Spells
 {
-    boss_huhuran(Creature* creature) : ScriptedAI(creature) { }
+    SPELL_FRENZY                = 26051, // triggers SPELL_POISON_BOLT
+    SPELL_BERSERK               = 26068, // triggers SPELL_POISON_BOLT
+    SPELL_NOXIOUS_POISON        = 26053,
+    SPELL_WYVERN_STING          = 26180,
+    SPELL_ACID_SPIT             = 26050,
+    SPELL_WYVERN_STING_DAMAGE   = 26233,
+    SPELL_POISON_BOLT           = 26052,
+    SPELL_HARD_ENRAGE           = 26662
+};
 
-    uint32 Frenzy_Timer;
-    uint32 Wyvern_Timer;
-    uint32 Spit_Timer;
-    uint32 PoisonBolt_Timer;
-    uint32 NoxiousPoison_Timer;
-    uint32 FrenzyBack_Timer;
+enum Events
+{
+    EVENT_FRENZY                = 1,
+    EVENT_WYVERN_STING          = 2,
+    EVENT_ACID_SPIT             = 3,
+    EVENT_NOXIOUS_POISON        = 4,
+    EVENT_HARD_ENRAGE           = 5
+};
 
-    bool Frenzy;
-    bool Berserk;
+struct boss_huhuran : public BossAI
+{
+    boss_huhuran(Creature* creature) : BossAI(creature, DATA_HUHURAN) { }
 
     void Reset() override
     {
-        Frenzy_Timer = urand(25000, 35000);
-        Wyvern_Timer = urand(18000, 28000);
-        Spit_Timer = 8000;
-        PoisonBolt_Timer = 4000;
-        NoxiousPoison_Timer = urand(10000, 20000);
-        FrenzyBack_Timer = 15000;
-
-        Frenzy = false;
-        Berserk = false;
+        _Reset();
+        berserk = false;
+        hardEnrage = false;
     }
 
-    void UpdateAI(uint32 diff) override
+    void EnterCombat(Unit* /*who*/) override
     {
-        //Return since we have no target
-        if (!UpdateVictim())
-            return;
-
-        //Frenzy_Timer
-        if (!Frenzy && Frenzy_Timer <= diff)
-        {
-            DoCast(me, SPELL_FRENZY);
-            Talk(EMOTE_FRENZY_KILL);
-            Frenzy = true;
-            PoisonBolt_Timer = 3000;
-            Frenzy_Timer = urand(25000, 35000);
-        }
-        else Frenzy_Timer -= diff;
+        events.ScheduleEvent(EVENT_FRENZY, urand(25000, 35000));
+        events.ScheduleEvent(EVENT_WYVERN_STING, urand(18000, 28000));
+        events.ScheduleEvent(EVENT_ACID_SPIT, 8000);
+        events.ScheduleEvent(EVENT_NOXIOUS_POISON, urand(10000, 20000));
+        events.ScheduleEvent(EVENT_HARD_ENRAGE, 300000);
+    }
 
-        // Wyvern Timer
-        if (Wyvern_Timer <= diff)
-        {
-            DoCastAOE(SPELL_WYVERNSTING);
-            Wyvern_Timer = urand(15000, 32000);
-        }
-        else Wyvern_Timer -= diff;
+    void JustDied(Unit* /*killer*/) override
+    {
+        _JustDied();
+    }
```

Comment:
Remove this, already done by BossAI::JustDied by default.

Reply:
good to know, ok